pkgs/development/libraries/s2geometry/0001-Replace-uint64-with-uint64_t-since-nothing-defines-u.patch

```diff
@@ -0,0 +1,108 @@
+From aeb66fbe4b1bb3d4a4eb2f706db732d2ddffaf1d Mon Sep 17 00:00:00 2001
```

Comment:
Is this patch available upstream and we can fetch it?

Reply:
No, I wrote it myself because it doesn't build without it (at least on aarch64), due to using `uint64`, which isn't part of the C++ standard and didn't appear to be defined anywhere. Bizarrely, they use `uint8_t`, `uint16_t`, and `uint32_t`, but `uint64` instead of `uint64_t`.

I think the `absl` subdirectory which contains the problematic code is a vendored copy of google's abseil cpp library, so it _might_ be possible to replace it with `abseil-cpp`, which is already packaged in nixpkgs, but I haven't studied it too closely yet, so I can't say for sure. Failing that, I will look into upstreaming the patch if possible.

Reply:
I checked. Upstream abseil-cpp has diverged to far from the vendored copy, so it can't be used as a drop-in replacement. I've reported the issue and added a comment to the nix expression with a link to the issue (https://github.com/google/s2geometry/issues/165).